pkg/installmanager/installmanager.go

```diff
@@ -1149,15 +1149,40 @@ func waitForProvisioningStage(provision *hivev1.ClusterProvision, m *InstallMana
 
 func (m *InstallManager) setupSSHKnownHost(knownHost string) error {
 
-	if err := os.MkdirAll("/root/.ssh", 0700); err != nil {
-		m.log.WithError(err).Error("error creating /root/.ssh directory")
+	// Add our potentially random UID to /etc/passwd so ssh works. Need to shell out here as
+	// the go libraries for user info appear to rely on /etc/passwd, which our user is not in.
+	// TODO: temporary work so we can use libvirt over ssh, this whole function can go once we're no longer doing that.
+	out, err := exec.Command("id", "-u").Output()
```

Comment:
I think you can get this info from the user package. https://golang.org/pkg/os/user/#Current

Reply:
It appeared to not work if the user is not in passwd. :( I tried this in one of the earlier attempts.